core/src/main/java/org/infinispan/interceptors/distribution/TxDistributionInterceptor.java

```diff
@@ -281,7 +280,7 @@ private Object handleTxWriteCommand(InvocationContext ctx, WriteCommand command,
       // see if we need to load values from remote sources first
       remoteGetBeforeWrite(ctx, command, key);
 
-      return invokeNextInterceptor(ctx, command);
+      return ctx.forkInvocationSync(command);
```

Comment:
No, this method returns an `Object`, that is the result of the rest of the chain, and callers expect to do some processing with that result/after the rest of the chain is done.